packages/nextjs/services/database/bg-imports/20250423_location_socials.ts

```diff
@@ -0,0 +1,101 @@
+import { db } from "../config/postgresClient";
+import { lower, users as schemaUsers } from "../config/schema";
+import * as dotenv from "dotenv";
+import { eq } from "drizzle-orm";
+import * as path from "path";
+
+dotenv.config({ path: path.resolve(__dirname, "../../../.env.development") });
+
+const BG_USERS_ENDPOINT = `${process.env.NEXT_PUBLIC_BG_BACKEND}/builders`;
+
+export type BGBuildersApiResponse = {
+  id: string;
+  socialLinks?: {
+    twitter?: string;
+    github?: string;
+    telegram?: string;
+    discord?: string;
+    instagram?: string;
+    email?: string;
+  };
+  role?: string;
+  function?: string;
+  creationTimestamp?: number;
+  builds?: { submittedTimestamp: number; id: string }[];
+  ens?: string;
+  status?: { text?: string; timestamp?: number };
+  reachedOut?: boolean;
+  scholarship?: boolean;
+  batch?: { number?: string; status?: string };
+  location?: string;
+  builderCohort?: { name?: string; id?: string; url?: string }[];
+  stream?: {
+    lastContract?: number;
+    cap?: string;
+    lastIndexedBlock?: number;
+    balance?: string;
+    streamAddress?: string;
+    frequency?: number;
+  };
+  [key: string]: any;
+}[];
+
+export async function importLocationAndSocials() {
+  try {
+    console.log("==== Importing Location and Socials ====");
+    console.log("Fetching data from BG firebase...");
+    const response = await fetch(BG_USERS_ENDPOINT);
+    const data: BGBuildersApiResponse = await response.json();
+    console.log(`Fetched ${data.length} users`);
+
+    await db.transaction(async tx => {
+      for (const builder of data) {
+        const userAddress = builder.id;
+        if (!userAddress) continue;
+        const existingUser = await tx.query.users.findFirst({
+          where: eq(lower(schemaUsers.userAddress), userAddress.toLowerCase()),
+        });
+
+        // Prepare socials mapping
+        const socials = {
+          socialTelegram: builder.socialLinks?.telegram || undefined,
+          socialX: builder.socialLinks?.twitter || undefined,
+          socialGithub: builder.socialLinks?.github || undefined,
+          socialInstagram: builder.socialLinks?.instagram || undefined,
+          socialDiscord: builder.socialLinks?.discord || undefined,
+          socialEmail: builder.socialLinks?.email || undefined,
```

Comment:
nit: all that `|| undefined` not needed

Reply:
fixed at https://github.com/BuidlGuidl/SpeedRunEthereum-v2/pull/148/commits/f87a9ac814b56bfc3b99680d170ef21c76ab4292